test/libyul/yulOptimizerTests/memoryStoreRemover/simple8.yul

```diff
@@ -0,0 +1,21 @@
+{
+    f()
+    function f() {
+        let x := 10
+        let y := 20
+        // will never be read, so can be removed
+        mstore(x, y)
+    }
+}
+// ----
+// step: memoryStoreRemover
+//
+// {
+//     f()
+//     function f()
+//     {
+//         let x := 10
+//         let y := 20
+//         pop(y)
```

Comment:
Hmm? Why only pop one of the arguments of `mstore?` Why pop at all and not just replace `mstore` with an empty block?

Reply:
Assuming some other stage removes pops which have side-effect free arguments, I think this step should just replace both arguments with pop.

Reply:
`pop` is needed for replacing `mstore(x, f()) -> pop(f())`. I should probably add a test for that.

Also it makes things a bit easier, since we relace `FunctionCall` with another `FunctionCall`.

Reply:
This test has `mstore(x, y) -> pop(y)`. Likewise all the other tests just replace `mstore` with a single `pop` for the value, but not the offset.

Reply:
So the step modifies `mstore` of the form `mstore(x, expression)`. The key has to be a SSA identifier. The expression can be anything, including function calls with side effects. So we really need `pop(expression)`. The reason I didn't include `pop(x)` is because it's redundant, and also it's 'easier' to implement this way. The reason is that

```cpp
void operator()(FunctionCall& _funCall) 
{
    // Replacing _funCall by pop(expression) is possible here.
    // But _funCall by pop(key) and pop(expression) can only be done in `operator()(Block& _block)`. Slightly inconvenient.
}
```

Reply:
```
/// A helper class that would replace `mstore(key, expression)` for all `key` in
/// `m_keys`, by `pop(expression)`.
class EraseMemoryStores: public ASTModifier
```

This explains it well, and I see you check for `arg(0) == Identifier`. I guess I only jumped onto the tests, since there are so many changes.